packages/remirror__core/src/builtins/upload-extension/file-placeholder-actions.ts

```diff
@@ -15,4 +16,19 @@ interface RemovePlaceholderAction {
   id: string;
 }
 
-export type PlaceholderPluginAction = AddPlaceholderAction | RemovePlaceholderAction;
+interface UpdatPlaceholderAction {
+  type: ActionType.UPDATE_PLACEHOLDER;
+  id: string;
+  payload: any;
+}
+
+export type PlaceholderPluginAction =
+  | AddPlaceholderAction
+  | RemovePlaceholderAction
+  | UpdatPlaceholderAction;
+
+export interface PlaceholderPluginMeta {
+  added: Array<Omit<AddPlaceholderAction, 'type'>>;
+  removed: Array<Omit<RemovePlaceholderAction, 'type'>>;
+  updated: Array<Omit<UpdatPlaceholderAction, 'type'>>;
+}
```

Comment:
If `PlaceholderPluginMeta` is just an array of `PlaceholderPluginAction`, it seems both the typing could be simpler (e.g. no `Omit` is needed anymore).

Reply:
Even though the type property in `PlaceholderPluginMeta` doesn't have much use, keeping it directly can make the code look a bit simpler. I'll have a try.